src/lib.rs

```diff
@@ -264,8 +325,10 @@ pub struct Core {
     pub vendor: Vendor,
     #[doc = " CPU microarchitecture for this core"]
     pub uarch: Uarch,
-    #[doc = " Value of CPUID leaf 1 EAX register for this core"]
-    pub cpuid: u32,
+    #[doc = " Value of CPUID leaf 1 EAX register for this core (x86-specific ID)"]
+    pub cpuid: Option<u32>,
+    #[doc = " Value of Main ID Register (MIDR) for this core (arm-specific ID)"]
```

Comment:
aarch64? Not sure about 32-bit (which is typically what `arm` refers to)

Reply:
Looks like both: https://developer.arm.com/documentation/100442/0100/register-descriptions/aarch32-system-registers/midr--main-id-register